src/ansys/acp/core/_recursive_copy.py

```diff
@@ -0,0 +1,246 @@
+        return
+    visited_objects[key] = tree_object
+    graph.add_node(key)
+
+    if options.include_children:
+        for child_object in child_objects(tree_object):
+            if not isinstance(child_object, CreatableTreeObject):
+                continue
+            graph.add_edge(child_object._resource_path.value, key)
+            _build_dependency_graph_impl(
+                tree_object=child_object,
+                graph=graph,
+                visited_objects=visited_objects,
+                options=options,
+            )
+    if options.include_linked_objects:
+        for linked_object in all_linked_objects(tree_object):
+            assert isinstance(linked_object, CreatableTreeObject)
+            graph.add_edge(tree_object._resource_path.value, linked_object._resource_path.value)
+            _build_dependency_graph_impl(
+                tree_object=linked_object,
+                graph=graph,
+                visited_objects=visited_objects,
+                options=options,
+            )
+
+
+def recursive_copy(
+    *,
+    source_objects: Iterable[CreatableTreeObject],
+    parent_mapping: Iterable[tuple[TreeObject, TreeObject]],
+    include_children: bool = True,
+    include_linked_objects: bool = True,
+) -> list[CreatableTreeObject]:
+    """Recursively copy a tree of ACP objects.
+
+    This function copies a tree of ACP objects, starting from the given source objects.
+    You can specify whether to include children (for example the Modeling Plies in a
+    Modeling Group) and linked objects (for example the Rosettes linked to an Oriented
+    Selection Set) in the copy.
+
+    To specify where the new objects should be stored, you must provide a list of tuples
+    in the ``parent_mapping`` argument. Each tuple contains the original parent object
+    as the first element and the new parent object as the second element.
+    Note that this mapping may need to contain parent objects that are not direct parents
+    of the source objects, if another branch of the tree is included via linked objects.
+
+    The ``parent_mapping`` argument can also include objects which are part of the
+    ``source_objects`` list. In this case, the function will not create a new object for
+    the parent, but will use the existing object instead.
+
+    The function returns a list of newly created objects.
+
+    .. note::
+
+        Only attributes supported by PyACP are copied to the new objects.
+
+    Parameters
+    ----------
+    source_objects :
+        The starting point of the tree to copy.
+    parent_mapping :
+        A list of tuples defining where the new objects are stored. Each tuple contains
+        the original parent object as the first element and the new parent object as the
+        second element.
+    include_children :
+        Whether to include child objects when creating the tree to copy.
+    include_linked_objects :
+        Whether to include linked objects when creating the tree to copy.
```

Comment:
Remove these parameters, since it's not quite clear yet how exactly this should work / whether it is needed.